c10/core/build.bzl

```diff
@@ -18,6 +18,27 @@ def define_targets(rules):
         alwayslink = True,
     )
 
+    rules.cc_library(
+        name = "MapAllocator",
+        srcs = ["MapAllocator.cpp"],
+        hdrs = ["MapAllocator.h"],
+        linkstatic = True,
+        local_defines = [
+            "C10_BUILD_MAIN_LIB",
+            "HAVE_MMAP=1",
+            "HAVE_SHM_OPEN=1",
+            "HAVE_SHM_UNLINK=1",
+        ],
+        visibility = ["//visibility:public"],
+        deps = [
+            ":CPUAllocator",
+            ":base",
+            "//c10/util:base",
+        ],
+        linkopts = ["-lrt"],
```

Comment:
This is not actually a valid argument for buck rules.
Also this is not portable and rt should be pulled in as a deps.